staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -482,6 +482,13 @@ func (c *Cacher) Delete(
 // Watch implements storage.Interface.
 func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions) (watch.Interface, error) {
 	pred := opts.Predicate
+	// If the resourceVersion is unset, ensure that the rv
+	// from which the watch is being served, is the latest
+	// one. "latest" is ensured by serving the watch from
+	// the underlying storage.
+	if opts.ResourceVersion == "" {
+		return c.storage.Watch(ctx, key, opts)
```

Comment:
This is technically much more expensive request, but given there should be very few of those and eventually we can optimize it, I'm fine with it for now.

Reply:
Gotcha!
When you say optimise it, do you mean in terms of further reducing the number of times we actually have to call into etcd by changing the if condition?